### ISSUE TITLE
[NetCore 0.11.0] AzureRM.NetCore\Get-AzureRMResource -ResourceId without APIVersion fails

### ISSUE THREAD

**Issue by raghushantha** [Azure/azure-powershell#5938]
### Description

[NetCore 0.11.0] AzureRM.NetCore\Get-AzureRMResource -ResourceId without APIVersion fails

### Script/Steps for Reproduction

```powershell
Import-Module -Name AzureRM.NetCore
Get-Module -Name AzureRM.NetCore | fl -f

Name              : AzureRM.NetCore
Path              : C:\Program Files\PowerShell\Modules\AzureRM.NetCore\0.11.0\AzureRM.Netcore.psm1
Description       : [PowerShell .Net Core] Azure Resource Manager Module
ModuleType        : Script
Version           : 0.11.0

$DebugPreference = 'Continue'
Get-AzureRmResource -ResourceId "subscriptions/$subId/resourceGroups"
DEBUG: 5:21:51 PM - GetAzureResourceCmdlet begin processing with ParameterSet 'GetByResourceId'.
DEBUG: 5:21:51 PM - using account id '========================'...
Get-AzureRmResource : One or more errors occurred. (The type initializer for 'ApiVersionCache' threw an exception.)
At line:1 char:1
+ Get-AzureRmResource -ResourceId "subscriptions/$subId/resourceGroups"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Get-AzureRmResource], AggregateException
+ FullyQualifiedErrorId : System.AggregateException,Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.GetAzureResourceCmdlet

# Using APIVersion explicitly works fine
 Get-AzureRmResource -ResourceId "subscriptions/$subId/resourceGroups" -ApiVersion 2018-02-01
<works>
```

### Module Version

```powershell
Get-Module -Name AzureRM.NetCore -ListAvailable


    Directory: C:\Program Files\PowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     0.11.0     AzureRM.Netcore
Manifest   0.10.0     AzureRM.Netcore
```

### Environment Data

```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0-preview.1
PSEdition                      Core
GitCommitId                    v6.1.0-preview.1
OS                             Microsoft Windows 10.0.14393
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Debug Output

```powershell
$DebugPreference = 'Continue'
$subId = <your subscription Id after auth to Azure>
Get-AzureRmResource -ResourceId "subscriptions/$subId/resourceGroups"
DEBUG: 5:21:51 PM - GetAzureResourceCmdlet begin processing with ParameterSet 'GetByResourceId'.
DEBUG: 5:21:51 PM - using account id '========================'...
Get-AzureRmResource : One or more errors occurred. (The type initializer for 'ApiVersionCache' threw an exception.)
At line:1 char:1
+ Get-AzureRmResource -ResourceId "subscriptions/$subId/resourceGroups"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Get-AzureRmResource], AggregateException
+ FullyQualifiedErrorId : System.AggregateException,Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.GetAzureResourceCmdlet

```

**Comment by mrvladis** [comment]
Got  the same issue.

**Comment by MiYanni** [comment]
You can't do that call in the current (6.0) release of Azure PowerShell (for desktop), so it won't be supported in the next release for NetCore. Instead, you'd call:

```powershell
Get-AzureRmResourceGroup
```

That will list all the resource groups from the current context. However, I have tested a few different calls for both `Get-AzureRmResource` and `Get-AzureRmResourceGroup` and can no longer produce the issue. So, this will be fixed in the 0.12.0 NetCore release.

**Comment by mosharafMS** [comment]
I got the same issue on this code snippet on this line
`$nicObject=Get-AzureRmResource -ResourceId $vmNic.id | Get-AzureRmNetworkInterface `

The same script is working on the PS desktop module but doesn't work on core module. 
Do we have a timeline for the fix to be released? 

```
$vmNICs=$vm.NetworkProfile.NetworkInterfaces 
foreach($vmNic in $vmNICs)
{
    $nicObject=Get-AzureRmResource -ResourceId $vmNic.id | Get-AzureRmNetworkInterface 
    $ipconfigs=$nicObject.IpConfigurations
    $NSG=$nicObject.NetworkSecurityGroup
    Remove-AzureRmNetworkInterface -Name $nicObject.Name -ResourceGroupName $ResourceGroupName -Force
    Remove-AzureRmResource -ResourceId $ipconfigs.PublicIpAddress.Id -Force 
    Remove-AzureRmResource -ResourceId $NSG.Id -Force 
}
```

**Comment by MiYanni** [comment]
@raghushantha @mrvladis @mosharafMS I'm currently trying to get the 0.12.0 build of the NetCore modules merged. Please keep in mind that Azure PowerShell for .NET Core is in preview. We are currently in the process of reviewing my PR for the changes. This `Get-AzureRmResource` issue will be fixed in that PR. You can track the PR here: https://github.com/Azure/azure-powershell/pull/6281

**Comment by markcowl** [comment]
Fixed in the release candidate

**Comment by MiYanni** [comment]
@raghushantha @mrvladis @mosharafMS Fixed in [0.12.0 for NetCore](https://github.com/Azure/azure-powershell/releases/tag/v0.12.0-nc-June2018)
